### PR TITLE
Fix moving of file into library

### PR DESCRIPTION
**Pull request by koppor**
Fixes https://github.com/JabRef/jabref-issue-melting-pot/issues/630

Follow-up to #11576

Additionally:

- We added support for modifier keys when dropping a file on an entry in the main table.
- When dropping a file into the main table, after copy or move, the file is now renamed according to the configured patterns.

### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Need to test under Mac, there was something about modifiers not working

**Comment by koppor** [comment]
> Need to test under Mac, there was something about modifiers not working

Test is drag and drop with and without modifier keys>

- outside file directory
- inside file directory

Which means: 6 different manual test cases.

You can also add drag'n'drop from another volume, which makes then 9 test cases.

File: src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java [hunk 1]
```diff
@@ -51,13 +52,30 @@ public Optional<Path> copyFileToFileDir(Path file) {
         Optional<Path> firstExistingFileDir = bibDatabaseContext.getFirstExistingFileDir(filePreferences);
         if (firstExistingFileDir.isPresent()) {
             Path targetFile = firstExistingFileDir.get().resolve(file.getFileName());
+            if (file.equals(targetFile)) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java:55]: Equals on Path ca be misleading we probably want ot check the filename only

File: src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java [hunk 2]
```diff
@@ -51,13 +52,30 @@ public Optional<Path> copyFileToFileDir(Path file) {
...
     }
 
+    public void copyLinkedFilesToFileDir(BibEntry entry) {
+        List<LinkedFile> files = entry.getFiles();
+        for (LinkedFile file : files) {
+            LinkedFileHandler fileHandler = new LinkedFileHandler(file, entry, bibDatabaseContext, filePreferences);
+            try {
+                fileHandler.copyToDefaultDirectory();
+            } catch (IOException exception) {
+                LOGGER.error("Error while moving file {}", file.getLink(), exception);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java:73]:
```suggestion
                LOGGER.error("Error while copying file {}", file.getLink(), exception);
```

**Comment by Siedlerchr** [comment]
Mac only supports copy https://bugs.openjdk.org/browse/JDK-8264172

File: src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java]: 59f38fb10a85e8a5e1a0519c2fe9f2539586ba78

**Comment by InAnYan** [comment]
I've only done 3 test cases with a .bib file outside of the PDFs files directory.

The PR works!

**Comment by koppor** [comment]
One needs to check whether the file indexing works - or causes trouble... I would bet it works, because I first handle the file on the file system and then add it to jabref which then indexes it...

**Comment by LoayGhreeb** [comment]
Fixed compilation errors, removed `IndexManager` from `ImportHandler`, and removed the index blocker.
Tested, the indexing works without issues.

**Pull request merged**